scripts/sign_encrypt.py

```diff
@@ -29,391 +59,876 @@ def int_parse(str):
+
+            offs += 4;
+            if img_type == SHDR_ENCRYPTED_TA:
+                ehdr = self.inf[offs: offs + EHDR_SIZE]
+                offs += EHDR_SIZE
+                [enc_algo, flags, nonce_len,
+                 tag_len] = struct.unpack('<IIHH', ehdr)
+
+                enc_algo_name = 'Unkown'
+                if enc_algo in enc_tee_alg.values():
+                    enc_algo_name = value_to_key(enc_tee_algo, enc_algo)
+                print('  enc_algo:   {} ({})'.format(enc_algo, enc_algo_name))
+
+                if enc_algo not in enc_tee_alg.values():
+                    raise Exception('Unrecognized encrypt algorithm: 0x{:08x}'
+                                    .format(enc_value))
+                print('  nonce_len:  {} (bytes)'.format(nonce_len))
+                if nonce_len != NONCE_SIZE:
+                    raise Exception("Unexpected nonce len: {}"
+                                    .format(nonce_len))
+                nonce = self.inf[offs:offs + nonce_len]
+                print('  nonce:      {}...'.format(binascii.hexlify(nonce)))
+                offs += nonce_len
+
+                print('  tag_len:    {} bytes'.format(tag_len))
+                if tag_len != TAG_SIZE:
+                    raise Exception("Unexpected tag len: {}".format(tag_len))
+                tag = self.inf[-tag_len:]
+                print('  tag:        {}...'.format(binascii.hexlify(tag)))
+                ciphertext = self.inf[offs:-tag_len]
+                print('  Ciph offset:{} (0x{:x}) bytes)'.format(offs, offs))
+                print('  Ciph size:  {} (0x{:x}) (bytes)'
+                      .format(len(ciphertext), len(ciphertext)))
+                if len(ciphertext) != img_size:
+                    raise Exception("Unexpected ciphertext size: ",
+                                    "got {}, expected {}"
+                                    .format(len(ciphertext), img_size))
+            else:
+                img = self.inf[offs:]
+                print('  TA offset:  {} (0x{:x}) bytes'.format(offs, offs))
+                print('  TA size:    {} (0x{:x}) bytes'
+                      .format(len(img), len(img)))
+                if len(img) != img_size:
+                    raise Exception("Unexpected img size: got {}, expected {}"
+                                    .format(len(img), img_size))
+            offs += img_size
+
+
+        offs = 0
+        while offs < len(self.inf):
+            if offs > 0:
+                # name_size is the previous subkey header
+                name_img = self.inf[offs:offs + name_size]
+                print('  next name:  "{}"'.format(name_img_to_str(name_img)))
+                offs += name_size
+                print('Next header at offset: {} (0x{:x})'
+                      .format(offs, offs))
+
+            shdr = self.inf[offs:offs + SHDR_SIZE]
+            [magic, img_type, img_size, algo_value, digest_len,
+             sig_len] = struct.unpack('<IIIIHH', shdr)
```

Comment:
`s/digest_len/hash_size/` and `s/sig_len/sig_size/` to match the C definition and the text below